sdk/v1/client_test.go

```diff
@@ -190,25 +190,120 @@ func TestClient_PostJob(t *testing.T) {
+		Convey("When search-reindexClient.PostJob is called", func() {
 			job, err := searchReindexClient.PostJob(ctx, client.Headers{})
 			So(err, ShouldNotBeNil)
 			So(err.Error(), ShouldEqual, "failed as unexpected code from search reindex api: 409")
 			So(apiError.ErrorStatus(err), ShouldEqual, http.StatusConflict)
 
-			Convey("then the expected empty job is returned", func() {
+			Convey("Then the expected empty job is returned", func() {
 				So(job, ShouldResemble, models.Job{})
 			})
 
-			Convey("and client.Do should be called once with the expected parameters", func() {
+			Convey("And client.Do should be called once with the expected parameters", func() {
 				doCalls := httpClient.DoCalls()
 				So(doCalls, ShouldHaveLength, 1)
 				So(doCalls[0].Req.URL.Path, ShouldEqual, path)
 			})
 		})
 	})
 }
+
+func TestClient_PostTasksCount(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	testJobID := "883c81fd-726d-4ea3-9db8-7e7c781a01cc"
+	pathToCheck := "v1/jobs/883c81fd-726d-4ea3-9db8-7e7c781a01cc/tasks"
+
+	mockTaskToCreate := `{"task_name":"zebedee","number_of_documents": "10"}`
+	testPayload := []byte(mockTaskToCreate)
+
+	Convey("Given clienter.Do doesn't return an error", t, func() {
+		expectedTask := models.Task{
+			JobID:       "883c81fd-726d-4ea3-9db8-7e7c781a01cc",
+			LastUpdated: time.Now().UTC(),
+			Links: &models.TaskLinks{
+				Self: "http://localhost:12150/jobs/abc123/tasks/zebedee",
+				Job:  "http://localhost:12150/jobs/abc123",
+			},
+			NumberOfDocuments: 10,
+			TaskName:          "zebedee",
+		}
+
+		body, err := json.Marshal(expectedTask)
+		if err != nil {
+			t.Errorf("failed to setup test data, error: %v", err)
+		}
+
+		httpClient := newMockHTTPClient(
+			&http.Response{
+				StatusCode: http.StatusCreated,
+				Body:       io.NopCloser(bytes.NewReader(body)),
+			},
+			nil)
+
+		searchReindexClient := newSearchReindexClient(t, httpClient)
+
+		Convey("When search-reindexClient.PostTasksCount is called", func() {
+			task, err := searchReindexClient.PostTasksCount(ctx, client.Headers{}, testJobID, testPayload)
+			So(err, ShouldBeNil)
+
+			Convey("Then the expected jobid is returned", func() {
+				So(testJobID, ShouldEqual, expectedTask.JobID)
+				So(task.NumberOfDocuments, ShouldEqual, expectedTask.NumberOfDocuments)
+			})
+
+			Convey("And client.Do should be called once with the expected parameters", func() {
+				doCalls := httpClient.DoCalls()
+				So(doCalls, ShouldHaveLength, 1)
+				So(doCalls[0].Req.URL.Path, ShouldEqual, pathToCheck)
+			})
+		})
+	})
+	Convey("Given a 500 response", t, func() {
+		httpClient := newMockHTTPClient(&http.Response{StatusCode: http.StatusInternalServerError}, nil)
+		searchReindexClient := newSearchReindexClient(t, httpClient)
+
+		Convey("When search-reindexClient.PostTaskCount is called", func() {
+			task, err := searchReindexClient.PostTasksCount(ctx, client.Headers{}, testJobID, testPayload)
```

Comment:
This will need to be changed then:
```suggestion
			task, err := searchReindexClient.PostTasksCount(ctx, client.Headers{}, testJobID, mockTaskToCreate)
```

Reply:
not implemented in line with my other comments as above